app/controllers/users/approvals_controller.rb

```diff
@@ -22,4 +26,15 @@ def reject
     render "users/approvals/approve"
   end
 
+  def check_user
+    unless current_user?(params[:user_id])
+      flash[:notice] = "You are signed in as #{current_user.username}"
+      if params[:origin]
+        developer = Developer.find_by(company_name: params[:company_name])
```

Comment:
This is what I tried to do but the request that sends the user to coposition is currently a redirect_to (so you can't set the headers). I tried replacing the redirect with an HTTParty get request but then WFAT complained about the controller not finding the correct route. It does feel like all requests should have the API-KEY header though so I might have another go.